Calendar/app/src/main/java/com/elysium/calendar/MainActivity.java

```diff
@@ -0,0 +1,105 @@
+package com.elysium.calendar;
+
+import android.Manifest;
+import android.content.Context;
+import android.database.Cursor;
+import android.provider.CalendarContract;
+import android.provider.ContactsContract;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.LoaderManager;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.CursorLoader;
+import android.support.v4.content.Loader;
+import android.support.v4.widget.CursorAdapter;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+import static android.content.pm.PackageManager.PERMISSION_DENIED;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+public class MainActivity extends AppCompatActivity implements LoaderManager.LoaderCallbacks<Cursor> {
+
+    public static final int PERMISSIONS_REQUEST_CODE = 88;
+    public static final int CALENDAR_LOADER = 0;
+
+    private RecyclerView mRecycler;
+    private CursorAdapter mAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        if (ContextCompat.checkSelfPermission(this,
+                Manifest.permission.READ_CONTACTS)
+                != PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.READ_CALENDAR, Manifest.permission.WRITE_CALENDAR},
+                    PERMISSIONS_REQUEST_CODE);
+        }
+
+        mRecycler = (RecyclerView)findViewById(R.id.recycler_view);
+        mRecycler.setLayoutManager(new LinearLayoutManager(this));
+
+        mAdapter = new CursorAdapter(new ArrayList<ContactsContract.CommonDataKinds.Event>()) {
```

Comment:
`CursorAdapter` works with `ListView` but not with `RecyclerView`. You should not use it here. Instead, instantiate your custom RecyclerView Adapter class.